### PR TITLE
Fix extensions printing

### PR DESCRIPTION
**Pull request by davesnx**
When printing code with refmt: Extensions are printed with and without brackets `[]` in some cases, creating a few problems: 

- Different expressions are treated differently when formatting (that might give the impression to be different expressions when they aren't)
- Creates unbalance code where used inside branches

This PR tries to unify them to always add `[]` while respecting the inline let attributes added https://github.com/reasonml/reason/pull/1703.

### CODE REVIEW

File: dune [hunk 1]
```diff
@@ -1 +1 @@
-(ignored_subdirs (node_modules js))
+(data_only_dirs node_modules js)
```
Comment (anmonteiro) [inline dune:1]: this should be `(dirs :standard \ node_modules js)`.

`data_only_dirs` is a different concept.

File: formatTest/typeCheckedTests/expected_output/attributes.re [hunk 2]
```diff
@@ -237,7 +237,10 @@ class boxB ('a) (init: 'a) =
 
 /* To be able to put an attribute on just the return in that case, use
  * parens. */
-[@onBoxC x; y]
+[@onBoxC
```
Comment (anmonteiro) [inline formatTest/typeCheckedTests/expected_output/attributes.re:240]: this one is a little awkward

File: formatTest/typeCheckedTests/expected_output/attributes.rei [hunk 3]
```diff
@@ -78,7 +78,5 @@ type ttttt = {
   x: int,
 };
 
-module Foo: {
-  [@someattr]
-  let foo: int => int;
-};
+module Foo: {[@someattr]
```
Comment (anmonteiro) [inline formatTest/typeCheckedTests/expected_output/attributes.rei:81]: this needs to be fixed

File: formatTest/unit_tests/expected_output/extensions.re [hunk 4]
```diff
@@ -1,13 +1,12 @@
 /* Extension sugar */
 
-%extend
```
Comment (anmonteiro) [inline formatTest/unit_tests/expected_output/extensions.re:3]: if this was valid syntax, shouldn't we preserve its printing?

Comment (anmonteiro) [reply formatTest/unit_tests/expected_output/extensions.re:3]: I guess this is what the PR is fixing, ignore my comment.

**Review by anmonteiro — COMMENTED** [review]
looks good, there's just the module case to fix

File: formatTest/typeCheckedTests/expected_output/attributes.re (hunk 2 above)
Comment (davesnx) [reply formatTest/typeCheckedTests/expected_output/attributes.re:240]: At first, I found it weird but later made sense. Any structure_item inside an attribute should be treated like a structure_item without an attribute...

I can roll it back thought.

File: formatTest/typeCheckedTests/expected_output/attributes.rei (hunk 3 above)
Comment (davesnx) [reply formatTest/typeCheckedTests/expected_output/attributes.rei:81]: Fixed in e17fe1b

Comment (anmonteiro) [reply formatTest/typeCheckedTests/expected_output/attributes.rei:81]: very nice, thanks

File: formatTest/typeCheckedTests/expected_output/attributes.re (hunk 2 above)
Comment (anmonteiro) [reply formatTest/typeCheckedTests/expected_output/attributes.re:240]: that does make sense. happy to keep it as is

**Review by anmonteiro — APPROVED** [review]
LGTM. I think we're ready to land this

**Pull request merged**